File: umbrella-insurance-core/src/main/java/com/umbrella/insurance/core/models/entities/Roster.java
package com.umbrella.insurance.core.models.entities;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

@Getter
@Setter
@Entity
@Table(name = "rosters", schema = "public")
public class Roster {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "roster_id", nullable = false)
    private Long id;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "person_id")
    private Person person;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "team_id")
    private Team team;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "season_id")
    private Season season;

    @Column(name = "roster_name", length = Integer.MAX_VALUE)
    private String rosterName;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "team_member_type_id")
    private TeamMemberType teamMemberType;

}

File: umbrella-insurance-react-ui/src/models/rosters/v1/Roster.ts
import { Person } from "../../people/v1/Person";
import { Season } from "../../seasons/v1/Season";
import { TeamMemberType } from "../../teamMemberTypes/v1/TeamMemberType";
import { Team } from "../../teams/v1/Team";

export class Roster {
    id?: number;
    team?: Team;
    person?: Person;
    season?: Season;
    rosterName?: string;
    teamMemberType?: TeamMemberType;

    constructor(obj?:any) {
        if(obj) {
            if(obj?.id) {
                this.id = obj.id;
            }
            if(obj?.team) {
                this.team = new Team(obj.team);
            }
            if(obj?.person) {
                this.person = new Person(obj.person);
            }
            if(obj?.season) {
                this.season = new Season(obj.season);
            }
            if(obj?.rosterName) {
                this.rosterName = obj.rosterName;
            }
            if(obj?.teamMemberType) {
                this.teamMemberType = new TeamMemberType(obj.teamMemberType);
            }
        }
    }
}